### PR TITLE
Tests for new dataset APIs

### PR DESCRIPTION
**Pull request by hhsecond**
## Motivation and Context
Tests for new dataset APIs

### CODE REVIEW

File: tests/ml_datasets/test_dataset.py [hunk 1]
```diff
@@ -156,6 +163,47 @@ def test_multiple_dataset_batched_loader(self, repo_20_filled_samples):
...
+        # testing with batch_size = 1
+        dset = make_numpy_dataset([first_aset, second_aset], batch_size=1, drop_last=True)
+        total_samples = 0
+        for dset1, dset2 in dset:
+            total_samples += dset1.shape[0]
+            assert dset1.shape == (1, 5, 7)
+            assert dset2.shape == (1, 5, 7)
+        assert total_samples == 20  # drop last is True will not have any effect
+
+        with pytest.raises(RuntimeError):
```
Comment (rlizzo) [inline tests/ml_datasets/test_dataset.py:176]: Use `match='foo message'` in order to make sure the correct error is raised here.

File: tests/ml_datasets/test_dataset.py [hunk 2]
```diff
@@ -156,6 +163,47 @@ def test_multiple_dataset_batched_loader(self, repo_20_filled_samples):
...
+            # Setting drop_last without batching
+            dset = make_numpy_dataset([first_aset, second_aset], batch_size=0, drop_last=True)
+        dset = make_numpy_dataset([first_aset, second_aset], batch_size=0)
+        total_samples = 0
+        for dset1, dset2 in dset:
+            total_samples += 1
+            assert dset1.shape == (5, 7)
+            assert dset2.shape == (5, 7)
+        assert total_samples == 20
```
Comment (rlizzo) [inline tests/ml_datasets/test_dataset.py:185]: Need a test where `drop_last=True` actually has an effect.

**Review by rlizzo — CHANGES_REQUESTED** [review]
Two minor test additions which should be included. Otherwise looks good!

File: tests/ml_datasets/test_dataset.py (hunk 2 above)
Comment (hhsecond) [reply tests/ml_datasets/test_dataset.py:185]: That was already there. On line #165

File: tests/ml_datasets/test_dataset.py (hunk 1 above)
Comment (hhsecond) [reply tests/ml_datasets/test_dataset.py]: Done

**Pull request merged**